test/tlv/TlvArray.ts

```diff
@@ -0,0 +1,53 @@
+/**
+ * @license
+ * Copyright 2022 Project CHIP Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { TlvArray } from "../../src/tlv/TlvArray.js";
+import { TlvString } from "../../src/tlv/TlvString.js";
+import { ByteArray } from "../../src/util/ByteArray.js";
+
+type TestVector<I, E> = {[testName: string]: { input: I, out: E }};
+
+const validateTestVector: TestVector<string[], boolean> = {
+    "validates an array with an acceptable length": { input: ["a", "b"], out: false },
+    "throws an error if the array is too short": { input: ["a"], out: true },
+    "throws an error if the array is too long": { input: ["a", "b", "c", "d", "e"], out: true },
+};
+
+describe("TlvArray", () => {
+    const schema = TlvArray(TlvString());
+
+    describe("encode", () => {
+        it("encodes an array", () => {
+            const result = schema.encode(["a", "b", "c"]);
+
+            expect(result.toHex()).toBe("160c01610c01620c016318");
+        });
+    });
+
+    describe("decode", () => {
+        it("decodes an array", () => {
+            const result = schema.decode(ByteArray.fromHex("160c01610c01620c016318"));
```

Comment:
I'd suggest adding a codec test vector and adding a few more tests (arrays of various types perhaps).

Reply:
Added in PR #20